src/enumerable.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    i = 0
+    selected_arr = []
+    while i < length
+      selected_arr << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    selected_arr
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return false if yield(self[i]) == false || yield(self[i]).nil?
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    while i < length
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_none?
+    i = 0
+    while i < length
+      return true if yield(self[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_count
+    i = 0
+    count = []
+
```

Comment:
- [ ] my_count counts the number of items in enum that are equal to item if an argument is given